File: BaekJoon/Silver/10845/Python/10845.py
from collections import deque
import sys

n = int(sys.stdin.readline())
queue1 = deque()
for i in range(0, n):
    cmd = sys.stdin.readline().rstrip()
    if 'push' in cmd:
        num = cmd.split().pop()
        queue1.append(num)
    elif cmd == 'front':
        if not queue1:
            print(-1)
        else:
            front_value = queue1.popleft()
            queue1.appendleft(front_value)
            print(front_value)
    elif cmd == 'back':
        if not queue1:
            print(-1)
        else:
            back_value = queue1.pop()
            queue1.append(back_value)
            print(back_value)
    elif cmd == 'pop':
        if not queue1:
            print(-1)
        else:
            print(queue1.popleft())
    elif cmd == 'size':
        print(len(queue1))
    else:
        print(1 if not queue1 else 0)

File: BaekJoon/Silver/10845/C++/10845.cpp
// no.10845: 큐 (S4)

#include <cstdio>
#include <deque>
#include <cstring>
using namespace std;

int main() {
    int n;
    scanf("%d", &n);
    deque<int> deq;
    for(int i=0; i<n; i++) {
        char inst[20];
        scanf("%s", inst);
        if(strcmp(inst, "push")==0) {
            int x;
            scanf("%d", &x);
            deq.push_back(x);
        }
        if(strcmp(inst, "pop")==0) {
            if(deq.empty()) {
                printf("-1\n");
                continue;
            }
            printf("%d\n", deq.front());
            deq.pop_front();
        }
        if(strcmp(inst, "size")==0) {
            printf("%lu\n", deq.size());
        }
        if(strcmp(inst, "empty")==0) {
            printf("%d\n", deq.empty() ? 1 : 0);
        }
        if(strcmp(inst, "front")==0) {
            if(deq.empty()) {
                printf("-1\n");
                continue;
            }
            printf("%d\n", deq.front());
        }
        if(strcmp(inst, "back")==0) {
            if(deq.empty()) {
                printf("-1\n");
                continue;
            }
            printf("%d\n", deq.back());
        }
    }
    return 0;
}